cpp/src/io/comp/uncomp.cpp

```diff
@@ -350,12 +356,27 @@ gdf_error io_uncompress_single_h2d(const void *src, gdf_size_type src_size, int
                     cdfh_ofs += cdfh_len;
                 }
             }
-            if (strm_type != IO_UNCOMP_STREAM_TYPE_INFER)
-                break; // Fall through for INFER
-        default:
-            // Unsupported format
-            break;
         }
+        if (strm_type != IO_UNCOMP_STREAM_TYPE_INFER)
+            break; // Fall through for INFER
+    case IO_UNCOMP_STREAM_TYPE_BZIP2:
+        if (src_size > 4)
+        {
+            const bz2_file_header_s *fhdr = (const bz2_file_header_s *)raw;
+            // Check for BZIP2 file signature "BZh1" to "BZh9"
+            if (fhdr->sig[0] == 'B' && fhdr->sig[1] == 'Z' && fhdr->sig[2] == 'h' && fhdr->sig[3] >= '1' && fhdr->sig[3] <= '9')
```

Comment:
For the `BZh` part, why not std::string::compare() or strncmp for clarity/brevity?

Reply:
I personally prefer individual checks for 3-characters on something that is not a zero-terminated string (Ideally would have liked a 32-bit word signature with if (sig >= 'BZh1' && sig <= 'BZh9') but some compilers complain that multi-character constants are non-standard (presumably due to implied assumption about endianness).